src/ngraph/op/fused/group_conv.cpp

```diff
@@ -49,23 +51,46 @@ op::GroupConvolution::GroupConvolution(const Output<Node>& data_batch,
     constructor_validate_and_infer_types();
 }
 
+op::GroupConvolution::GroupConvolution(const Output<Node>& data_batch,
+                                       const Output<Node>& filters,
+                                       const Strides& window_movement_strides,
+                                       const Strides& window_dilation_strides,
+                                       const CoordinateDiff& padding_below,
+                                       const CoordinateDiff& padding_above,
+                                       const Strides& data_dilation_strides,
+                                       const PadType& pad_type)
+    : FusedOp({data_batch, filters})
+    , m_window_movement_strides(window_movement_strides)
+    , m_window_dilation_strides(window_dilation_strides)
+    , m_padding_below(padding_below)
+    , m_padding_above(padding_above)
+    , m_data_dilation_strides(data_dilation_strides)
+    , m_groups(filters.get_partial_shape()[0])
+    , m_pad_type(pad_type)
+{
+    constructor_validate_and_infer_types();
+}
+
 void op::GroupConvolution::pre_validate_and_infer_types()
 {
     auto data_shape = get_input_partial_shape(0);
     auto filters_shape = get_input_partial_shape(1);
+
     if (data_shape.is_static() && filters_shape.is_static())
     {
         // Data channels
         NODE_VALIDATION_CHECK(this,
-                              data_shape.to_shape()[1] % m_groups == 0,
+                              data_shape.to_shape()[1] % get_groups() == 0,
```

Comment:
```suggestion
                              m_groups.is_dynamic() || data_shape.to_shape()[1] % static_cast<size_t>(m_groups) == 0,
```

Reply:
Similar change for all checks in this function.

Reply:
In this context `get_groups()` should return static value because `filters_shape` is static.
I wanted to avoid these `static_casts` so changed `m_groups` => `get_groups()`.
Does it make sense to you?

Reply:
Ah, I wasn't following that line of reasoning! In general the line of reasoning that "x came from y, and y is all static, therefore x is static" makes sense, but in this case I think it will not _quite_ work, because `m_groups` was set once at construction time, but `filters_shape` may have changed due to graph rewriting.

Here's an idea:

- always set `m_groups` to `Dimension::dynamic()` in the initializer list;
- update `m_groups` at the top of `pre_validate_and_infer_types`.

That way, if the input shape has changed, `m_groups` will be brought up to date once the node is revalidated.

Or, we could change `get_groups()` to query the filter shape, and select the relevant dimension, and not store `m_groups` at all. (But then you would need accessors for `size_t` as well as `Dimension`.)

Reply:
We need to store `m_groups` because groups can still be provided via attribute.
I think that updating `m_groups` at the top of `pre_validate_and_infer_types` is a good solution.
But one question:
Is it possible that after rewriting graph pass which was messing up with filters could update `m_groups` via public setter?

Reply:
> We need to store m_groups because groups can still be provided via attribute.

I see. That seems like an awkward design, but I suppose it's not a big deal.

> Is it possible that after rewriting graph pass which was messing up with filters could update `m_groups` via public setter?

Yes, it is possible. But in general any time we are mutating an attribute, all bets are off with respect to graph validity.

Reply:
Yep!